### PR TITLE
Add API for product actions with bulk operations

### PR DESCRIPTION
**Pull request by chrisvire**
Changed the `ProductAction` type so that the `types` property is now `actions` to be consistent.

Add the following APIs:

* Get actions/products for selected projects
`POST /api/product-action` `{ "project" : [ "int1", "int2" ... ], "id" : "guid" }`
=>
`{ "actionName1" : [ "guid1", "guid2", "guid3" ], "actionName2" : [ "guild4", "guid5" ] }`

* Run action on products
`POST /api/product-action/run` `{ "action" : "actionName", "products" : ["guid1", "guid2" ] }`

### CODE REVIEW

**Comment by chrisvire** [comment]
@NullVoxPopuli: I got a decent start on the Service and Controller.  However, I am not sure how to get `List<int>` or `List<Product>` to `ProjectsController.GetProductActions([FromBody] List<Project> projects)`.  I created `GetProductTest.Get_ProductActions_For_Projects()` to try to get all the projects (for the current user) and then pass those in as the content for a POST, but that didn't work.

**Comment by chrisvire** [comment]
@NullVoxPopuli Please review.  I believe this is ready to be merged.

File: source/OptimaJet.DWKit.StarterApplication/BackendServicesExtension.cs [hunk 1]
```diff
@@ -48,6 +48,8 @@ public static IServiceCollection AddApiServices(this IServiceCollection services
                 {
                     builder.AddResource<ProductActions,Guid>("product-actions");
                     builder.AddResource<ProjectToken>("project-tokens");
+                    builder.AddResource<ProductActionRuns, Guid>("product-action-runs");
```
Comment (NullVoxPopuli) [inline source/OptimaJet.DWKit.StarterApplication/BackendServicesExtension.cs]: do these need to be resources if they aren't jsonapi payloads? (or are they?)

File: source/OptimaJet.DWKit.StarterApplication/Models/ProductActions.cs [hunk 2]
```diff
@@ -6,7 +6,22 @@ namespace OptimaJet.DWKit.StarterApplication.Models
...
+    public class ProductActionRuns : Identifiable<Guid>
+    {
+        [Attr("action")]
+        public string Action { get; set; }
+
+        [Attr("products")]
+        public List<string> Products { get; set; }
+    }
+
+    public class ProductActionProjects : Identifiable<Guid>
```
Comment (NullVoxPopuli) [inline source/OptimaJet.DWKit.StarterApplication/Models/ProductActions.cs]: Maybe: `ProjectIdList`?
idk, ProductActionProjects confuses me :shrug:

File: source/OptimaJet.DWKit.StarterApplication/Models/ProductActions.cs [hunk 3]
```diff
@@ -6,7 +6,22 @@ namespace OptimaJet.DWKit.StarterApplication.Models
 {
     public class ProductActions : Identifiable<Guid>
     {
-        [Attr("types")]
-        public List<string> Types { get; set; }
+        [Attr("actions")]
+        public List<string> Actions { get; set; }
+    }
+
+    public class ProductActionRuns : Identifiable<Guid>
```
Comment (NullVoxPopuli) [inline source/OptimaJet.DWKit.StarterApplication/Models/ProductActions.cs]: `ProductActionRuns` tells me that these are things that have been run.

Maybe `AvailableProductsForAction`?

File: source/OptimaJet.DWKit.StarterApplication/BackendServicesExtension.cs (hunk 1 above)
Comment (chrisvire) [reply source/OptimaJet.DWKit.StarterApplication/BackendServicesExtension.cs]: They were until last night when I figured out the issues.  I removed them and the tests still pass.  I will change this.

File: source/OptimaJet.DWKit.StarterApplication/Models/ProductActions.cs (hunk 2 above)
Comment (chrisvire) [reply source/OptimaJet.DWKit.StarterApplication/Models/ProductActions.cs]: Changed to ProductActionProjectIds

File: source/OptimaJet.DWKit.StarterApplication/Models/ProductActions.cs (hunk 3 above)
Comment (chrisvire) [reply source/OptimaJet.DWKit.StarterApplication/Models/ProductActions.cs]: Changed to ProductActionRunRequest

**Pull request merged**